pkg/apis/aws/v1alpha1/awsfederatedaccountaccess_types.go

```diff
@@ -0,0 +1,102 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AWSFederatedAccountAccessState defines the various status an FederatedAccountAccess CR can have
+type AWSFederatedAccountAccessState string
+
+const (
+	// AWSFederatedAccountAccessStateInProgress const for InProgress status state
+	AWSFederatedAccountAccessStateInProgress AWSFederatedAccountAccessState = "InProgress"
```

Comment:
I know they're different types, but do we really need two vars per CRD just to have the string "InProgress"? Same for "Applied" and "Failed"

Reply:
So the conditions need a conditiontype for the way that conditions work through our utils. I do agree its a lot of the same , but i was just following the pattern.